### ISSUE TITLE
[I] Some settings are applied even if you click on the "Cancel" button

### ISSUE THREAD

**Issue by Katarn** [pbek/QOwnNotes#2481]
#### Expected behaviour

The editor's color scheme should not be applied when clicking "Cancel" in the settings dialog.

#### Actual behaviour

The editor's color scheme is applied when the application is restarted, even if the "Cancel" button is clicked in the settings dialog.

#### Steps to reproduce

1. Open settings dialog.
2. Change editor's color scheme in color schemes list.
3. Click to Cancel button.
4. Restart the app.

Perhaps some other settings apply too. I propose to check in the source code not only the color scheme of the editor.

#### Information about the application, settings and environment

<details><summary>Expand</summary>

QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `Ср мар 2 15:28:29 2022`
**Version**: `22.2.10`
**Build date**: `Feb 27 2022`
**Build number**: `954`
**Platform**: `windows`
**Operating System**: `Windows 8.1 Version 6.3 (Build 9600)`
**Build architecture**: `x86_64`
**Current architecture**: `x86_64`
**Release**: `GitHub Actions`
**Qt Version (build)**: `5.15.2`
**Qt Version (runtime)**: `5.15.2`
**Portable mode**: `no`
**Qt Debug**: `0`
**Locale (system)**: `ru_RU`
**Locale (interface)**: *empty*
**Primary screen resolution**: `1920x1080`
**Screen resolution(s)**: `1920x1080, 1920x1080`
**Icon theme**: `breeze-dark-qownnotes`
**Notes in current note folder**: `254`
**Calendar items**: `0`
**Enabled scripts**: `2`
</details>

**Comment by pbek** [comment]
> The editor's color scheme is applied when the application is restarted, even if the "Cancel" button is clicked in the settings dialog.

TL;DR: Yes 😁 

Some settings need to be applied immediately, it would be very hard to keep track of them.
